thrift/thrift.bzl

```diff
@@ -78,6 +78,7 @@ rm -rf {out}_tmp"""
     thrift = struct(
       srcs = ctx.outputs.libarchive,
       transitive_srcs = transitive_srcs,
+      external_jars = ctx.attr.external_jars,
```

Comment:
I think you need to keep the transitive union of external_jars to dependencies work correctly. See what is happening with `_collect_thrift_srcs`.

also while at it, can we use `depset` vs `set`. `set` is a deprecated alias for `depset`:
https://docs.bazel.build/versions/master/skylark/lib/set.html